File: fixtures/scale/scala/large/src/cryptobench/ciphers/speck/SpeckKey.scala
package cryptobench.ciphers.speck

final case class SpeckKey(k0: Int, k1: Int, k2: Int, k3: Int) {

  def k(index: Int): Int = (index & 3) match {
    case 0 => k0
    case 1 => k1
    case 2 => k2
    case _ => k3
  }
}

object SpeckKey {
  def default(): SpeckKey = SpeckKey(0x01234567, -0x76543211, -0x1234568, 0x76543210)
}

File: fixtures/scale/java/large/src/cryptobench/ciphers/speck/SpeckKey.java
package cryptobench.ciphers.speck;

public class SpeckKey {
    private final int[] words;

    public SpeckKey(int k0, int k1, int k2, int k3) {
        this.words = new int[] { k0, k1, k2, k3 };
    }

    public int k(int index) {
        return words[index & 3];
    }

    public static SpeckKey defaultKey() {
        return new SpeckKey(0x0123_4567, 0x89AB_CDEF, 0xFEDC_BA98, 0x7654_3210);
    }
}
